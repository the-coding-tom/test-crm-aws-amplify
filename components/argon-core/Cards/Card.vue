<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
