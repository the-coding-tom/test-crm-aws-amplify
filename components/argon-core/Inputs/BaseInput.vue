<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
