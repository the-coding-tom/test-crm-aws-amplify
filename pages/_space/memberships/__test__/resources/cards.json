{
  "data": [
    {
      "id": "ccof:Ksaklsalksaklsak",
      "card_brand": "MASTERCARD",
      "last_4": "9399",
      "exp_month": 6,
      "exp_year": 2022,
      "cardholder_name": "Al Ksal"
    }
  ]
}
