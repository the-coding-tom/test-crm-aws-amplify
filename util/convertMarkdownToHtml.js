export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
