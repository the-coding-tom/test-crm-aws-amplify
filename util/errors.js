export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
