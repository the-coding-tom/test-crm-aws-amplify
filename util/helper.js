const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
