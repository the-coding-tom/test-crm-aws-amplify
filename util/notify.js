const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
