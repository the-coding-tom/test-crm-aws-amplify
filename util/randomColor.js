const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
