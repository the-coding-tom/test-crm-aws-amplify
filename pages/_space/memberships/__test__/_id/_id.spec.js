import MemberPage from '@/pages/_space/memberships/_id/index.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'
import Card from '~/components/argon-core/Cards/Card.vue'
import Badge from '~/components/argon-core/Badge.vue'
import Vue2Filters from 'vue2-filters'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import lodash from 'lodash'
import moment from 'moment'

import cards from '@/pages/_space/memberships/__test__/resources/cards.json'
import currentSpace from '@/pages/_space/__test__/resources/current-space.json'
import allMembers from '@/pages/_space/memberships/__test__/resources/all-memberships-uninvited.json'

describe('Page: /:space/memberships/_id/ | Memberships Id Page', () => {
  let NuxtStore, store, wrapper, $membership, router

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $membership = {
      getPaymentMethods: jest.fn().mockResolvedValue(cloneDeep(cards)),
      getAMembership: jest.fn().mockResolvedValue({ data: allMembers[1] }),
      deleteMembership: jest.fn().mockResolvedValue({ data: {} }),
      updateMembership: jest.fn().mockResolvedValue()
    }

    global._ = lodash
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: []
    })

    const params = {
      id: allMembers[1].id
    }

    const asyncDataResponse = await MemberPage.asyncData({
      store,
      $membership,
      params
    })

    wrapper = createWrapper(
      MemberPage,
      store,
      router,
      {
        loading: false,
        admins: [],
        options: [
          { text: 'Referral', value: 'referral' },
          { text: 'Application', value: 'application' }
        ],
        paid_by: null,
        prefix_type: [
          {
            text: 'Founding Member',
            value: '0'
          },
          {
            text: 'Fast Track',
            value: '1'
          },
          {
            text: 'Early Invite',
            value: '2'
          },
          {
            text: 'General Member',
            value: '3'
          }
        ],
        prefix_locality: [
          {
            text: 'Bay Area',
            value: 'A'
          },
          {
            text: 'US outside of the Bay',
            value: 'B'
          },
          {
            text: 'EMEA',
            value: 'C'
          },
          {
            text: 'APAC',
            value: 'D'
          }
        ],
        searching: false,
        cards: [],
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'no-ssr': true,
          'client-only': true,
          transition: false,
          'b-popover': true,
          'date-picker': true
        },
        mocks: {
          $membership: $membership,
          $moment: moment
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.use(Vue2Filters)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(Card.name, Card)
        vueInstance.component(Badge.name, Badge)
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('cancel member', async () => {
    // mock confirm
    global.confirm = jest.fn().mockReturnValue(true)

    wrapper.find('button.btn.text-danger.btn-transparent').trigger('click')
    await flushPromises()

    expect($membership.deleteMembership).toHaveBeenCalled()
  })

  it('renders member information', async () => {
    expect(wrapper.find('#email').element.value).toBe(allMembers[1].email)
    expect(wrapper.find('#firstName').element.value).toBe(
      allMembers[1].first_name
    )
    expect(wrapper.find('#lastName').element.value).toBe(
      allMembers[1].last_name
    )
    expect(wrapper.find('#membershipNumber').element.value).toBe(
      allMembers[1].membership_number
    )
    expect(wrapper.find('#linkedinURL').element.value).toBe(
      allMembers[1].linkedin_url
    )
    expect(wrapper.find('#comment').element.value).toBe(
      allMembers[1].extras[0].comment
    )
    expect(wrapper.find('#email').element.value).toBe(allMembers[1].email)
  })

  it('updates member', async () => {
    const updateMemberInfo = {
      firstName: 'Lata',
      lastName: 'Paul'
    }

    wrapper.find('#lastName').setValue(updateMemberInfo.lastName)
    wrapper.find('#firstName').setValue(updateMemberInfo.firstName)
    await flushPromises()

    wrapper.find('form').trigger('submit.prevent')
    await flushPromises()

    expect($membership.updateMembership).toHaveBeenCalled()
    expect($membership.updateMembership.mock.calls[0][1].first_name).toBe(
      updateMemberInfo.firstName
    )
    expect($membership.updateMembership.mock.calls[0][1].last_name).toBe(
      updateMemberInfo.lastName
    )
  })
})
