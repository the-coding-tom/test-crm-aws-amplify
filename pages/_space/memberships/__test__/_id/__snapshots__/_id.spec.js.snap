// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/memberships/_id/ | Memberships Id Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between align-items-center py-4">
            <h1 class="page-title"><span>Membership</span> <b>â€¢</b> Leslie Akorful</h1>
            <div><button type="button" class="btn text-primary btn-transparent"><i class="fas fa-user-lock"></i> Brivo Onboard
              </button> <button type="button" class="btn text-danger btn-transparent"><i class="fa fa-trash"></i> Cancel Membership</button></div>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="row">
        <div class="col-md-6">
          <div class="card">
            <!---->
            <div class="card-body">
              <div class="row">
                <div class="form-group col-md-6"><label class="form-control-label">
                    Email
                  </label>
                  <div class="input-group has-label">
                    <!----> <input type="text" valid="true" required="required" id="email" disabled="disabled" placeholder="Email" class="form-control">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
                <div class="form-group col-md-6"><label class="form-control-label">
                    First Name
                  </label>
                  <div class="input-group has-label">
                    <!----> <input type="text" valid="true" required="required" id="firstName" placeholder="First Name" class="form-control">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
                <div class="form-group col-md-6"><label class="form-control-label">
                    Last Name
                  </label>
                  <div class="input-group has-label">
                    <!----> <input type="text" valid="true" required="required" id="lastName" placeholder="Last Name" class="form-control">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
                <div class="form-group col-md-6"><label class="form-control-label">
                    Membership #
                  </label>
                  <div class="input-group has-label">
                    <!----> <input type="text" valid="true" required="required" id="membershipNumber" placeholder="0000000" class="form-control">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
                <div class="form-group col-md-6"><label class="form-control-label">
                    Linkedin
                  </label>
                  <div class="input-group has-label">
                    <!----> <input type="text" valid="true" id="linkedinURL" placeholder="https://linkedin.com/in/someone" class="form-control">
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
                <fieldset class="form-group col-md-6" id="__BVID__14">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__14__BV_label_">Source of Invitation</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="invitationSource" class="custom-select">
                      <option value="referral">Referral</option>
                      <option value="application">Application</option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__16">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__16__BV_label_">Comments</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring"><textarea id="comment" placeholder="Comments" rows="2" wrap="soft" class="form-control" label="Comments"></textarea>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__18">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__18__BV_label_">Membership Type</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="memberType" class="custom-select">
                      <option value="0">Founding Member</option>
                      <option value="1">Fast Track</option>
                      <option value="2">Early Invite</option>
                      <option value="3">General Member</option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__20">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__20__BV_label_">Region</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="region" class="custom-select">
                      <option value="A">Bay Area</option>
                      <option value="B">US outside of the Bay</option>
                      <option value="C">EMEA</option>
                      <option value="D">APAC</option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__22">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__22__BV_label_">Assigned Admin</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring">
                    <div class="el-select">
                      <!---->
                      <div class="el-input el-input--suffix">
                        <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose an admin" class="el-input__inner">
                        <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                        <!----></span>
                        <!---->
                        <!---->
                      </div>
                      <div class="el-select-dropdown el-popper" style="display: none;">
                        <div class="el-scrollbar" style="display: none;">
                          <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                            <ul class="el-scrollbar__view el-select-dropdown__list"></ul>
                          </div>
                          <div class="el-scrollbar__bar is-horizontal">
                            <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                          </div>
                          <div class="el-scrollbar__bar is-vertical">
                            <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                          </div>
                        </div>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__30">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__30__BV_label_">Start Date</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring">
                    <client-only-stub>
                      <date-picker-stub id="time" width="100%" input-class="form-control" lang="en" format="YYYY-MM-DD" value-type="format" confirm="" type="date" placeholder="Start Date" value=""></date-picker-stub>
                    </client-only-stub>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__33" aria-describedby="__BVID__33__BV_description_">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__33__BV_label_">Trial Days</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring"><input type="number" required="required" min="0" aria-required="true" class="form-control" id="__BVID__34">
                    <!---->
                    <!----><small tabindex="-1" class="form-text text-muted" id="__BVID__33__BV_description_">Number of days before member is charged</small></div>
                </fieldset>
                <fieldset class="form-group col-md-6" id="__BVID__35">
                  <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__35__BV_label_">Paid for</legend>
                  <div tabindex="-1" role="group" class="bv-no-focus-ring">
                    <div class="custom-control custom-checkbox"><input type="checkbox" autocomplete="off" class="custom-control-input" value="true" id="__BVID__36"><label class="custom-control-label" for="__BVID__36">Yes</label></div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
                <!---->
                <div class="col-md-12"><button type="submit" class="btn btn-primary btn-block">Save</button></div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
        </div>
      </div>
    </div>
  </form> <button type="button" class="btn text-primary ml-2 mb-2 btn-transparent"><i class="fas fa-chevron-left"></i> Back</button>
</div>
`;
